src/main/java/io/supertokens/inmemorydb/Start.java

```diff
@@ -1619,4 +1627,86 @@ public void addInfoToNonAuthRecipesBasedOnUserId(String className, String userId
             throw new IllegalStateException("ClassName: " + className + " is not part of NonAuthRecipeStorage");
         }
     }
+
+    // TOTP recipe: 
+
```

Comment:
all of these queries below need to detect various sql failures and return the right exceptions..

For example, the query for `markDeviceAsVerified`, if the row was not updated, then you need to throw a `UnknownDeviceException`.

Reply:
Fiex.